Devscord.DiscordFramework/Integration/Services/DiscordClientRolesService.cs

```diff
@@ -57,12 +63,20 @@ public async Task SetRolePermissions(IEnumerable<ChannelContext> channels, Disco
         {
             await Task.Delay(1000);
             var createdRole = this.GetSocketRoles(server.Id).FirstOrDefault(x => x.Id == role.Id);
-            var channelPermissions = new OverwritePermissions(permissions.AllowPermissions.GetRawValue(), permissions.DenyPermissions.GetRawValue());
+            if (createdRole == null)
+            {
+                Log.Error("Created role (name: {roleName}) was null", role.Name);
+                return;
+            }
 
-            Parallel.ForEach(channels, async c =>
+            var channelPermissions = new OverwritePermissions(permissions.AllowPermissions.GetRawValue(), permissions.DenyPermissions.GetRawValue());
+            Parallel.ForEach(channels, c =>
             {
-                var channelSocket = (IGuildChannel) await this._discordClientChannelsService.GetChannel(c.Id);
-                await channelSocket.AddPermissionOverwriteAsync(createdRole, channelPermissions);
+                var channelSocket = (IGuildChannel)this._discordClientChannelsService.GetChannel(c.Id).Result;
+                if (channelSocket != null && channelSocket.PermissionOverwrites.All(x => x.TargetId != createdRole.Id))
+                {
+                    channelSocket.AddPermissionOverwriteAsync(createdRole, channelPermissions).Wait();
```

Comment:
czytałeś o tym Wait() czy po prostu zmieniłeś?

Reply:
logiczniej się wydaje synchronicznie
ale nie mogę za bardzo nic znaleźć o tym co jest lepszym rozwiązaniem
Choć w tym przypadku to i tak małe znaczenie bo tych channels najczęściej jest kilkanaście, więc to niezbyt duża liczba przełączeń